docs/proposals/kubemark.md

```diff
@@ -0,0 +1,164 @@
+or it can be reconstructed from Events.
+
+When we have Hollow Node ready, we’ll be able to test performance of Master Components by creating a real Master Node, with API server, Controllers, etcd and whatnot, and
+create number of Hollow Nodes that will register to the running Master.
+
+For the initial version we will teach Master components to use port number to identify Kubelet/KubeProxy. This will allow running those components on non-default ports, and in
+the same time will allow to run multiple Hollow Nodes on a single machine. During setup we will generate credentials for cluster communication and pass them to
+HollowKubelet/HollowProxy to use. Master will treat all HollowNodes as normal ones.
+
+As a second (equivalent) option we will run Kubemark on top of 'real' Kubernetes cluster, where both Master and Hollow Nodes will be Pods.
+In this option we'll be able to use Kubernetes mechanisms to streamline setup, e.g. by using Kubernetes networking to ensure unique IPs for
+Hollow Nodes, or using Secrets to distribute Kubelet credentials. The down side of this configuration is that it's likely that some noise
+will appear in Kubemark results from either CPU/Memory preassure from other things running on Nodes (e.g. FluentD, or Kubelet) or running
+cluster over an overlay network. We belive that it'll be possible to turn off cluster monitoring for Kubemark runs, so that the impact
+of real Node daemons will be minimized, but we don't know what will be the impact of using higher level networking stack. Running a
+compariston will be an interesting test in itself.
+
+To make Kubemark easier to maintain when system evolves Hollow components will reuse real "production" code for Kubelet and KubeProxy, but will mock all the backends with no-op or very simple mocks. We belive that this approach is better in the long run than writing special "performance-test-aimed" separate version of them. This may take more time to create an initial version, but we think maintenance cost will be noticeably smaller.
+
+![Kubmark architecture diagram](Kubemark_architecture.png?raw=true "Kubemark architecture overview")
+
+## Things to simulate
+
+In real Kubernetes on a single Node we run two daemons that communicate with Master in some way: Kubelet and KubeProxy. In this paragraph we list what kind of communication we
+have between those daemons and Master components. First step in creating holograms is understanding the communication between Master and Node components.
+
+### KubeProxy
+
+Creates two Watch clients, one for Services, one for Endpoints (it’s done in pkg/proxy/config/api.go, because why not). KubeProxy also exposes healthz, but it’s not read by
+the Master (or anything else except supervisord AFAICT). As a replacement for KubeProxy we'll use HollowProxy, which will be a real
+KubeProxy with injected no-op mocks everywhere it makes sense.
```

Comment:
Personally, I would just leave:
"As a replacement for KubeProxy we'll use HollowProxy, which will be a real KubeProxy with injected no-op mocks everywhere it makes sense."

The rest of that doesn't provide any value in my opinion.